R/forecast.R

```diff
@@ -426,6 +432,33 @@ new_forecast <- function(data, classname) {
 }
 
 
+#' Remakes a forecast object with a subclass
+#'
+#' This function takes a forecast object and modifies its class name by removing
+#' the old class name and adding a new class name. It also performs an assertion
+#' check on the modified forecast object.
+#'
+#' @param forecast The forecast object to be modified
+#' @param old_classname The name of the class to be removed from the forecast
+#' object
+#' @param new_classname The name of the class to be added to the forecast object
+#' @param verbose A logical value indicating whether to print verbose output
+#'
+#' @return The modified forecast object
+#' @keywords internal
+remake_forecast <- function(
+  forecast, old_classname, new_classname, verbose = TRUE
+) {
+  remade_forecast <- forecast
+  class(remade_forecast) <- setdiff(class(forecast), old_classname)
```

Comment:
`new_forecast()` calls `as.data.table()` which strips the class anyway

Reply:
I tried this and was having some weird errors. Can you confirm? More generally I think its helpful to make this explicit in the code.

Reply:
Hmm. I'll check it out and play around a bit with this. Circling back!